accounts-db/src/accounts.rs

```diff
@@ -552,31 +569,56 @@ impl Accounts {
         account_locks: &mut AccountLocks,
         writable_keys: Vec<&Pubkey>,
         readonly_keys: Vec<&Pubkey>,
-    ) -> Result<()> {
+        entry_accts_lookup: &mut EntryAcctLocks,
+        allow_self_conflicting_entries: &bool,
```

Comment:
just pass a `bool`, using a reference here is likely only slower and less convenient